ee/test/test_migration_0012.py

```diff
@@ -0,0 +1,94 @@
+        )
+        self.property_definition_without_tags = EnterprisePropertyDefinition.objects.create(
+            team_id=self.team.id, name="property def without tags",
+        )
+        TaggedItemModel.objects.create(tag=tag, property_definition_id=self.property_definition_with_tags.id)
+
+        # Setup for batched tags
+        self.team2_total_property_definitions = 2_300
+        tag2 = TagModel.objects.create(name="existing tag", team_id=self.team2.id)
+        with transaction.atomic():
+            for _tag in range(self.team2_total_property_definitions):
+                EnterprisePropertyDefinition.objects.create(
+                    name=f"batch_prop_{_tag}",
+                    team_id=self.team2.id,
+                    deprecated_tags=[_tag, "existing tag", "tag to create"],
+                )
+        TaggedItemModel.objects.create(
+            tag=tag2,
+            property_definition_id=EnterprisePropertyDefinition.objects.filter(team_id=self.team2.id).first().id,
+        )
+
+    def test_tags_migrated(self):
+        Tag = self.apps.get_model("posthog", "Tag")  # type: ignore
+        TaggedItem = self.apps.get_model("posthog", "TaggedItem")  # type: ignore
+        EnterpriseEventDefinition = self.apps.get_model("ee", "EnterpriseEventDefinition")  # type: ignore
+        EnterprisePropertyDefinition = self.apps.get_model("ee", "EnterprisePropertyDefinition")  # type: ignore
+
+        event_definition = EnterpriseEventDefinition.objects.get(id=self.event_definition.id)
+        self.assertEqual(event_definition.tagged_items.count(), 4)
+        self.assertEqual(
+            list(event_definition.tagged_items.order_by("tag__name").values_list("tag__name", flat=True)),
+            ["a", "b", "c", "existing tag"],
+        )
+
+        property_definition_with_tags = EnterprisePropertyDefinition.objects.get(
+            id=self.property_definition_with_tags.id
+        )
+        self.assertEqual(property_definition_with_tags.tagged_items.count(), 3)
+        self.assertEqual(
+            list(property_definition_with_tags.tagged_items.order_by("tag__name").values_list("tag__name", flat=True)),
+            ["c", "d", "existing tag"],
+        )
+
+        property_definition_without_tags = EnterprisePropertyDefinition.objects.get(
+            id=self.property_definition_without_tags.id
+        )
+        self.assertEqual(property_definition_without_tags.tagged_items.count(), 0)
+
+        # By the end of the migration, the total count for team 2 should be
+        # Tags = team2_total_property_definitions + 2 + team1_tags
+        # TaggedItems = team2_total_property_definitions * 3 + team1_taggeditems
+        self.assertEqual(Tag.objects.filter(team_id=self.team.id).all().count(), 5)
+        self.assertEqual(Tag.objects.all().count(), self.team2_total_property_definitions + 2 + 5)
+        self.assertEqual(TaggedItem.objects.all().count(), self.team2_total_property_definitions * 3 + 7)
```

Comment:
It would be the icing on the cake if we could also assert on the number of transactions that have occurred, but adding boilerplate to support that seems like scope creep. Any suggestions on how to do it easily are welcome!